apps/ejabberd/src/mod_mam_utils.erl

```diff
@@ -322,12 +322,12 @@ is_archivable_message(Mod, Dir, Packet=#xmlel{name = <<"message">>}) ->
 is_archivable_message(_, _, _) ->
     false.
 
-is_valid_message_type(_, _, <<"">>)          -> true;
-is_valid_message_type(_, _, <<"normal">>)    -> true;
-is_valid_message_type(_, _, <<"chat">>)      -> true;
-is_valid_message_type(_, _, <<"groupchat">>) -> true;
-is_valid_message_type(_, _, <<"error">>)     -> false;
-is_valid_message_type(_, _, _)               -> false.
+is_valid_message_type(_, _, <<"">>) -> true;
+is_valid_message_type(_, _, <<"normal">>) -> true;
+is_valid_message_type(_, _, <<"chat">>) -> true;
+is_valid_message_type(_, incoming, <<"groupchat">>) -> true;
```

Comment:
There is no point to archive outgoing message because at this point we can't be sure if the message reached the room and got broadcasted.